core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -415,21 +415,17 @@ class KafkaServer(val config: KafkaConfig, time: Time = SystemTime) extends Logg
 
   // Copy the subset of properties that are relevant to Logs
   // I'm listing out individual properties here since the names are slightly different in each Config class...
-  private def copyKafkaConfigToLog(serverProps: java.util.Map[String, Object]): java.util.Map[String, Object] = {
+  private def copyKafkaConfigToLog(kafkaConfig: KafkaConfig): java.util.Map[String, Object] = {
```

Comment:
I'm not sure at this point how best to handle this, but we need some testing in place for all derived configs.

It seems it should sit in `LogConfigTest`, but it does not add any value since this copy routine is what needs to be tested. We could consider having the `LogConfig` in `LogManagerTest` be created from this routine - i.e., instead of creating `LogConfig` directly, create `KafkaConfig`, then use this util (after making it `private[kafka]`) to copy it to the `LogConfig` that will be used in the `LogManagerTest`.